core/java/iesi-core/src/main/java/io/metadew/iesi/openapi/ConnectionParser.java

```diff
@@ -32,23 +36,30 @@ public static synchronized ConnectionParser getInstance() {
     public List<Connection> parse(OpenAPI openAPI) {
         String name = openAPI.getInfo().getTitle();
         String description = openAPI.getInfo().getDescription();
-        List<URL> adresses = getAdresses(openAPI.getServers());
-        return IntStream.range(0, adresses.size()).boxed()
+        Optional<Environment> environment = Optional.empty();
+        List<URL> addresses = getAdresses(openAPI.getServers());
+        try {
+            environment = EnvironmentConfiguration.getInstance().getAll().stream().findFirst();
+        } catch(RuntimeException exception) {
+
+        }
+
+        Optional<Environment> finalEnvironment = environment;
+        return IntStream.range(0, addresses.size()).boxed()
                 .map(index -> {
-                    URL address = adresses.get(index);
-                    String environment = String.format("env%s", index);
+                    URL address = addresses.get(index);
                     final List<ConnectionParameter> connectionParameters = new ArrayList<>();
                     getPort(address)
                             .ifPresent(port -> connectionParameters.add(
-                                    new ConnectionParameter(name, environment, "port", port)
+                                    new ConnectionParameter(name, finalEnvironment.map(Environment::getName).orElse(null), "port", port)
                             ));
                     getBaseUrl(address)
                             .ifPresent(baseUrl -> connectionParameters.add(
-                                    new ConnectionParameter(name, environment, "baseUrl", baseUrl)
+                                    new ConnectionParameter(name,finalEnvironment.map(Environment::getName).orElse(null), "baseUrl", baseUrl)
                             ));
-                    connectionParameters.add(new ConnectionParameter(name, environment, "host", getHost(address)));
-                    connectionParameters.add(new ConnectionParameter(name, environment, "tls", getProtocol(address)));
-                    return new Connection(name, "http", description, environment, connectionParameters);
+                    connectionParameters.add(new ConnectionParameter(name, finalEnvironment.map(Environment::getName).orElse(null), "host", getHost(address)));
+                    connectionParameters.add(new ConnectionParameter(name, finalEnvironment.map(Environment::getName).orElse(null), "tls", getProtocol(address)));
+                    return new Connection(name, "http", description, finalEnvironment.map(Environment::getName).orElse(null), connectionParameters);
 
                 }).collect(Collectors.toList());
     }
```

Comment:
is this still desired?
If no environments are found, the connection will use 'null' as environment key.
If the openapi documentation contains more than 1 urls, the respective connections will all have the same environment linked to it. This will cause a clash on DB level as (connection name, env name) is the primary key of a connection

Reply:
Indeed, I though I had deleted the pull request. It's better to do what you said during the last meeting.

Reply:
Okay, can you close the PR?